velox/exec/tests/OperatorUtilsTest.cpp

```diff
@@ -206,3 +207,54 @@ TEST_F(OperatorUtilsTest, gatherCopy) {
 TEST_F(OperatorUtilsTest, makeOperatorSpillPath) {
   EXPECT_EQ("spill/task_1_100", makeOperatorSpillPath("spill", "task", 1, 100));
 }
+
+TEST_F(OperatorUtilsTest, wrap) {
+  auto rowType = ROW({
+      {"bool_val", BOOLEAN()},
+      {"tiny_val", TINYINT()},
+      {"small_val", SMALLINT()},
+      {"int_val", INTEGER()},
+      {"long_val", BIGINT()},
+      {"ordinal", BIGINT()},
+      {"float_val", REAL()},
+      {"double_val", DOUBLE()},
+      {"string_val", VARCHAR()},
+      {"array_val", ARRAY(VARCHAR())},
+      {"struct_val", ROW({{"s_int", INTEGER()}, {"s_array", ARRAY(REAL())}})},
+      {"map_val",
+       MAP(VARCHAR(),
+           MAP(BIGINT(),
+               ROW({{"s2_int", INTEGER()}, {"s2_string", VARCHAR()}})))},
+  });
+
+  VectorFuzzer fuzzer({}, pool());
+  auto vector = fuzzer.fuzzFlat(rowType);
+  auto rowVector = vector->as<RowVector>();
+
+  for (int32_t iter = 0; iter < 20; ++iter) {
+    folly::Random::DefaultGenerator rng;
+    rng.seed(iter);
+    const int32_t wrapVectorSize =
+        iter == 0 ? 0 : 1 + folly::Random().rand32(rowVector->size(), rng);
```

Comment:
Any particular reason we are not testing indices buffer much larger than base vector?

Reply:
I thought I set the random base to be larger than the base vector. Thanks!

Reply:
> 1 + folly::Random().rand32(rowVector->size(), rng)

My reading is that wrapVectorSize will be at most the size of rowVector. Most likely less. Am I reading it wrong?

```
  /**
   * Returns a random uint32_t in [0, max). If max == 0, returns 0.
   */
  static uint32_t rand32(uint32_t max) {
```